File: test/combination/all_combinations.py
def printAllKLength(set, k): 
    n = len(set)  
    printAllKLengthRec(set, "", n, k) 
  
def printAllKLengthRec(set, prefix, n, k): 
    if (k == 0) : 
        f.write(prefix+"\n")
        #print(prefix)
        return
    for i in range(n): 
        newPrefix = prefix + set[i] 
        printAllKLengthRec(set, newPrefix, n, k - 1) 
  
if __name__ == "__main__": 
    f= open("all_combinations.txt","w+")
    ttt_set = ['x', 'o', '-'] 
    k = 9
    printAllKLength(ttt_set, k)

File: test/combination/all_combinations.go
package combination

import "fmt"

var (
	k      = 4
	values = []string{"x", "o", "-"}
)

func printAllKLength(set []string, k int) {
	n := len(set)
	printAllKLengthRec(set, "", n, k)
}

func printAllKLengthRec(set []string, prefix string, n int, k int) {
	if k == 0 {
		fmt.Println(prefix)
		return
	}
	for i := 0; i < n; i++ {
		newPrefix := prefix + set[i]
		printAllKLengthRec(set, newPrefix, n, k-1)
	}
}

func main() {
	//So much slower than python... Benchmark ?
	printAllKLength(values, k)
}
